integration/query_array_compat_test.go

```diff
@@ -0,0 +1,224 @@
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"DotNotationNoSuchFieldPosition": {
+			filter:     bson.D{{"v.some.0", bson.A{42}}},
+			resultType: emptyResult,
+		},
+		"DotNotationField": {
+			filter:        bson.D{{"v.array", int32(42)}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"DotNotationFieldPosition": {
+			filter:        bson.D{{"v.array.0", int32(42)}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"DotNotationFieldPositionQuery": {
+			filter:        bson.D{{"v.array.0", bson.D{{"$gte", int32(42)}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"DotNotationFieldPositionQueryNonArray": {
+			filter:     bson.D{{"v.document.0", bson.D{{"$lt", int32(42)}}}},
+			resultType: emptyResult,
+		},
+		"DotNotationFieldPositionField": {
+			filter:     bson.D{{"v.array.2.foo", "bar"}},
+			resultType: emptyResult,
+		},
+		"ElemMatchDoubleTarget": {
+			filter: bson.D{
+				{"_id", "double"},
+				{"v", bson.D{{"$elemMatch", bson.D{{"$gt", int32(0)}}}}},
+			},
+			resultType: emptyResult,
+		},
+		"ElemMatchGtZero": {
+			filter:        bson.D{{"v", bson.D{{"$elemMatch", bson.D{{"$gt", int32(0)}}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"ElemMatchGtZeroWithTypeArray": {
+			filter: bson.D{
+				{"v", bson.D{
+					{"$elemMatch", bson.D{
+						{"$gt", int32(0)},
+					}},
+					{"$type", "array"},
+				}},
+			},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"ElemMatchGtZeroWithTypeString": {
+			filter: bson.D{
+				{"v", bson.D{
+					{"$elemMatch", bson.D{
+						{"$gt", int32(0)},
+					}},
+					{"$type", "string"},
+				}},
+			},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"ElemMatchGtLt": {
+			filter: bson.D{
+				{"v", bson.D{
+					{"$elemMatch", bson.D{
+						{"$gt", int32(0)},
+						{"$lt", int32(43)},
+					}},
+				}},
+			},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"EqualityOne": {
+			filter:        bson.D{{"v", bson.A{int32(42)}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"EqualityTwo": {
+			filter:     bson.D{{"v", bson.A{42, "foo"}}},
+			resultType: emptyResult,
+		},
+		"EqualityThree": {
+			filter:        bson.D{{"v", bson.A{int32(42), "foo", nil}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"EqualityThree-reverse": {
+			filter:        bson.D{{"v", bson.A{nil, "foo", int32(42)}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"EqualityEmpty": {
+			filter:        bson.D{{"v", bson.A{}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"EqualityNull": {
+			filter:        bson.D{{"v", bson.A{nil}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"SizeFloat64": {
```

Comment:
Structuring idea: for some compat cases, we define multiple functions in the same file for different operations/operators. I find it helpful from the readability perspective. I wonder if we could do the same here. For example, everything `$size`-related could be in `TestQueryArrayCompatSize` or everything `$all`-related in `TestQueryArrayCompatAll`. As a referene, this is what we did in `query_logical_compat_test.go`, maybe we can do something similar here?

Reply:
I'll separated them, initially I thought they would be faster so I put them in the same func. But I just learnt that on CI it's faster to run multiple tests than one big test. Interesting.

Using multiple tests like `TestQueryArrayCompatSize`  `TestQueryArrayCompatAll` https://github.com/FerretDB/FerretDB/actions/runs/3529897317/jobs/5921499750**
```
ok  	github.com/FerretDB/FerretDB/integration	168.466s	coverage: 61.0% of statements in ../...
```
using one big test
https://github.com/FerretDB/FerretDB/actions/runs/3530990337/jobs/5923627738
```
ok  	github.com/FerretDB/FerretDB/integration	206.855s	coverage: 61.0% of statements in ../...
```

Reply:
@chilagrow, well, can we compare the time of two different CI runs? I'm not sure how GitHub Actions distribute resources, but I expect it might be possible that even the same test might take a different time to run on CI depends on available resources.

Reply:
true, never mind.